src/main/java/seedu/mtracker/filemanager/EtfDecoder.java

```diff
@@ -1,22 +1,55 @@
 package seedu.mtracker.filemanager;
 
+import seedu.mtracker.commons.Validate;
+import seedu.mtracker.error.FileLoadError;
 import seedu.mtracker.model.Instrument;
 import seedu.mtracker.model.InstrumentManager;
 import seedu.mtracker.model.subinstrument.Etf;
+import seedu.mtracker.ui.TextUi;
 
 public class EtfDecoder extends InstrumentDecoder {
 
     public static final int PAST_RETURNS_INDEX = 5;
     public static final int ETF_REMARKS_INDEX = 6;
+    protected static double decodedPastReturns;
+    protected static String decodedRemarks;
+
+    public static boolean isValidPastReturns(String[] textSegment) {
+        return Validate.isValidPastReturns(textSegment[PAST_RETURNS_INDEX]);
+    }
+
+    public static void validateAndDecodeSpecificAttributes(String[] textSegment) throws FileLoadError {
+        if (isValidPastReturns(textSegment)) {
+            decodeSpecificAttributes(textSegment);
+        }
+        throw new FileLoadError();
+    }
+
+    public static void tryValidateAndDecodeSpecificAttributes(String[] textSegment) {
+        try {
+            validateAndDecodeSpecificAttributes(textSegment);
+        } catch (Exception e) {
+            TextUi.showErrorMessage(e);
+        }
+    }
```

Comment:
The rest I think can follow the above. Basically need to create a few exceptions for each of the saved attribute having error. Then there isn't any need for FileLoadError <- I think FileLoadError should be used only used in loadFileMethod